app/models/aggregated_cash_income.rb

```diff
@@ -0,0 +1,106 @@
+class AggregatedCashIncome
+  include ActiveModel::Model
+
+  CATEGORIES= %i[
+      benefits
+      friends_or_family
+      maintenance_in
+      property_or_lodger
+      pension
+  ].freeze
+
+  RANGE = (1..3).freeze
+
+  def self.category_attributes(categories)
+    categories.each do |category|
+      attr_accessor "check_box_#{category}".to_sym
+      RANGE.each do |i|
+        attr_accessor "#{category}_month_#{i}".to_sym
+      end
+    end
+  end
+
+  attr_accessor :legal_aid_application_id,
+                :category,
+                :none_selected
+
+  category_attributes CATEGORIES
+
+  validate :check_category
+  validate :anything_selected?
+
+  def checkbox_integrity(params)
+    CATEGORIES.each do |category|
+      checkbox_selected = params["check_box_#{category}"] == 'true'
+
+      if checkbox_selected
+        @category = category
+        check_category
+      else
+        args = search_args(category)
+        CashTransaction.where(**args).destroy_all
+      end
+    end
+
+    anything_selected?
+  end
+
+  def save(params)
+    params.each do |key, value|
```

Comment:
I think this should be `def update(params)`.  We want to try and imitate the ActiveRecord API as much as possible, and this is doing what the `#update` method does.

Reply:
👍